dask/tests/test_utils.py

```diff
@@ -50,3 +54,59 @@ class Bar(object):
     assert foo(1.0) == 0.0
     assert foo(b) == b
     assert foo((1, 2.0, b)) == (2, 1.0, b)
+
+
+def test_nextlinesep():
+    lineseps = ('\r', '\n', '\r\n')
+    encodings = ('utf-16-le', 'utf-8')
+    for sep, encoding in product(lineseps, encodings):
+        euro = u'\u20ac'
+        yen = u'\u00a5'
+
+        bin_euro = u'\u20ac'.encode(encoding)
+        bin_yen = u'\u00a5'.encode(encoding)
+        bin_sep = sep.encode(encoding)
+
+        data = (euro * 10) + sep + (yen * 10) + sep + (euro * 10)
+        bin_data = data.encode(encoding)
+
+        with tmpfile() as fn:
+            with open(fn, 'w+b') as f:
+                f.write(bin_data)
+                f.seek(0)
+
+                start, stop = next_linesep(f, 5, encoding, sep)
+                assert start == len(bin_euro) * 10
+                assert stop == len(bin_euro) * 10 + len(sep.encode(encoding))
+
+                seek = len(bin_euro) * 10 + len(bin_sep) + len(bin_yen)
+                start, stop = next_linesep(f, seek, encoding, sep)
+
+                exp_start = len(bin_euro) * 10 + len(bin_sep) + len(bin_yen) * 10
+                exp_stop = exp_start + len(bin_sep)
+                assert start == exp_start
+                assert stop == exp_stop
+
+
+def test_gh606():
+    encoding = 'utf-16-le'
+    euro = u'\u20ac'
+    yen = u'\u00a5'
+    bin_euro = u'\u20ac'.encode(encoding)
+    bin_yen = u'\u00a5'.encode(encoding)
+
+    data = (euro * 10) + '\n' + (yen * 10) + '\n' + (euro * 10)
+    bin_data = data.encode(encoding)
+
+    with tmpfile() as fn:
+        with open(fn, 'w+b') as f:
+            f.write(bin_data)
+            f.seek(0)
+
+            stop = len(bin_euro) * 10 + len('\n'.encode(encoding))
+            res = textblock(f, 1, stop, encoding=encoding)
+            assert res == ((yen * 10) + '\n').encode(encoding)
+
+            stop = len(bin_euro) * 10 + len('\n'.encode(encoding))
+            res = textblock(f, 0, stop, encoding=encoding)
+            assert res == ((euro * 10) + '\n' + (yen * 10) + '\n').encode(encoding)
```

Comment:
@cowlicks do you have access to a Windows (virtual)machine?  If so can you check this test on that machine?

There are quite possibly a few machines floating around the Continuum office if you ask IT.